From 408c686e0d022926b486a524f2f2338042622c27 Mon Sep 17 00:00:00 2001
From: Poovendhan Selvaraj <quic_poovendh@quicinc.com>
Date: Thu, 14 Mar 2024 12:55:49 +0530
Subject: [PATCH] drivers: firmware: Check AES 128 bytes context SCM support is
 available

Add check to AES 128 bytes context SCM support is available before call
the scm call.

Signed-off-by: Poovendhan Selvaraj <quic_poovendh@quicinc.com>
Change-Id: I4e5ed414ca03b77f812dbbe2b63308996afec62e
---
 drivers/firmware/qcom_scm.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/firmware/qcom_scm.c b/drivers/firmware/qcom_scm.c
index d47e00b48a1a..364e570f4189 100644
--- a/drivers/firmware/qcom_scm.c
+++ b/drivers/firmware/qcom_scm.c
@@ -2883,7 +2883,13 @@ int __qti_scm_aes(struct device *dev, uint32_t req_addr,
 		.owner = ARM_SMCCC_OWNER_SIP,
 	};
 
-	ret = qcom_scm_call(__scm->dev, &desc, &res);
+	ret = __qcom_scm_is_call_available(__scm->dev, QTI_SVC_CRYPTO, cmd_id);
+	if (ret == 1) {
+		ret = qcom_scm_call(__scm->dev, &desc, &res);
+	} else {
+		pr_err("%s : Feature not supported by TZ..!\n", __func__);
+		return -EINVAL;
+	}
 
 	return res.result[0];
 }
-- 
2.34.1

